### DISCUSSION TITLE
Boot from SAN device failed: Error 0x3d222083

### DISCUSSION THREAD

**Discussion by EnikeevIlya** [ipxe/ipxe#1054] [Q&A] [answered]
Hello.

I have a problem with uploading a custom opnsense image via sanboot in iPXE.

Command:
`sanboot --no-describe http://192.168.180.74/opnsense-23.7.iso`

Error in IPXE 1.21.1 without debug:
`Boot from SAN device failed: Error 0x3d222083`

Error in IPXE 1.0.0 with DEBUG=efi_block:
`EFIBLK could not connect drivers: Error 0x7f22208e`

Could you tell me which drivers we are talking about? Is there something wrong with the iPXE build, the image, or the BIOS firmware?

Thank you very much in advance.

**Comment by NiKiZe** [comment, marked as answer]
I'd assume EFI firmware, have you used current iPXE git master?

Also, every error from iPXE includes a url to ipxe.org, make sure to include that in your post, and of course follow and read what that page has to say.

Have you booted this ISO on any other machine? Especially any with efi boot? Is the ISO efi compatible?

Also remember that any access to the ISO goes away once the kernel takes over, so everything needs to be loaded from the ISO in EFI space, squashfs images will not be accessible after boot.

**Reply by EnikeevIlya** [reply]
@NiKiZe Thank you very much, the problem was really with the EFi partition

**Comment by 3isenHeiM** [comment]
> @NiKiZe Thank you very much, the problem was really with the EFi partition


Hi @EnikeevIlya, I'm having this issue as well when trying to netboot an OPNsense ISO via iPXE.

Could you indicate me how you solve this issue ?

**Reply by NiKiZe** [reply]
Do you get the same http://ipxe.org/7f22208e error?
Does the ISO contain a EFI compatible structure as described on that error page?

For reference, what does `file yourfile.iso` show?

**Reply by 3isenHeiM** [reply]
Nope I got this error

```
EFIBLK 0x80 could not connect drivers: Error 0x7f31218e (https://ipxe.org/7f31218e)
```

```
# file filename.iso
filename.iso: ISO 9660 CD-ROM filesystem data 'OPNSENSE_261_AMD64'
```

**Reply by NiKiZe** [reply]
https://ipxe.org/7f31218e
So your UEFI firmware fails to boot. Exactly which ISO is this, and is it EFI compatible?
Have you tried to use the `.img` file instead?

**Reply by 3isenHeiM** [reply]
Yes I'm using the OPNsense vga img file. It's UEFI compatible.
I've managed to have it booting.
It loads correctly, but the OPNsense img has a 4th partition (UFS, labeled `OPNsense_Install`), and the installer needs to load it at boot.
Since the SAN “disk” that iPXE created is gone by the time the OPNsense kernel is loaded, it does not find this partition and fails to properly boot.

**Reply by NiKiZe** [reply]
Yes, True, you are back to
> Also remember that any access to the ISO goes away once the kernel takes over, so everything needs to be loaded from the ISO in EFI space, squashfs images will not be accessible after boot.
